routes/nodes.rb

```diff
@@ -0,0 +1,74 @@
+require 'sinatra/base'
+
+module Sinatra
+  module NodeRoutes
+    def self.registered(app)
+      ##
+      # Node Resource
+      ##
+
+      app.get '/node/new' do
+        # get new node form
+        erb :'nodes/edit'
+      end
+
+      app.get '/node/:id' do
+        # view a node
+        @node = NodeResource[id: params[:id]]
+        halt 404, 'node not found' if @node.nil?
+        erb :'nodes/show'
+      end
+
+      app.get '/node/:id/edit' do
+        # get node edit form
+        @node = NodeResource[id: params[:id]]
+        halt 404, 'node not found' if @node.nil?
+        erb :'nodes/edit'
+      end
+
+      app.get '/nodes' do
+        # get a list of all node
+        @nodes = NodeResource.all
+        erb :'nodes/index'
+      end
+
+      # This could also be PUT
+      app.post '/nodes' do
+        # recieve new node
+        node = NodeResource.create(project_id: Iam.settings.DB[:projects]
+                                                .where(name: params[:project_name])
+                                                .get(:id) || node.project_id,
```

Comment:
Does the `|| node.project_id` work if we're creating the `node` being referred to?